### PR TITLE
Added test reproducing a bug where consumable addon prevents from generating the next invoice

### PR DESCRIPTION
**Pull request by karser**
Hello @pierre ,
Please take a look at these 2 tests.
First one subscribes a user to a base monthly plan and checks that the invoice is created after 1 month.
Second subscribes a user to the same base plan and consumable addon and makes the same checks. For some reason, the addon prevents from creating the invoice. Recording usages doesn't make difference.

### CODE REVIEW

**Comment by karser** [comment]
Tests passed, that's odd. The second test from this PR fails on my local KB.
image

**Comment by karser** [comment]
I dropped my local database and tests now pass locally. But each test creates new tenant - they shouldn't interfere.
There is another KB instance deployed on a VPS with which tests fail against. if I drop the db, the tests will probably pass. Looks like a potential KB bug which prevents invoices generation.
Please tell me what do you think.

File: test/ServerInvoiceTest.php [hunk 1]
```diff
@@ -280,4 +280,77 @@ public function testCustomFields()
...
+
+        $invoices = $this->client->getAccountApi()->getInvoicesForAccount(
+            $this->account->getAccountId(),
+            null,
+            $withItems = 'true'
+        );
+        $this->assertCount(1, $invoices);
+        $this->assertSame(0.0, $invoices[0]->getAmount());
+
+        $this->clock->addDays(35);
```
Comment (pierre) [inline test/ServerInvoiceTest.php]: After moving the clock, asynchronous processing of subscriptions and invoice generation will happen.

We have some logic to ensure all asynchronous jobs are fully processed before returning (provided the [timeout](https://github.com/killbill/killbill/blob/6457b485fb32a182c76197a83f428123331f1380/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/TestResource.java#L198) is long enough), but the heuristic isn't always enough depending on the catalog, config, etc.

A good practice in these tests is to wait on the client side for an expected condition (e.g. a new invoice being generated) and fail the test if this doesn't happen after a timeout. See https://github.com/killbill/killbill-integration-tests/blob/2eec63809331bb0cb8217554542740352d5666aa/killbill-integration-tests/core/test_entitlement_create.rb#L29 for an example in our Ruby tests.

**Comment by karser** [comment]
I ported `wait_for_killbill` and `wait_for_expected_clause` methods.
It reduced the tests time on [CircleCI](https://circleci.com/gh/killbill/killbill-client-php/tree/pull%2F38) from 04:44 to 2:38. I guess that's the reason to merge this PR btw.

The tests still fail against my KB instance and invoices are not generated:
image

**Comment by pierre** [comment]
> The tests still fail against my KB instance and invoices are not generated:

Are you sure you don't have old data laying around? I suggest restarting your Kill Bill instance (to reset the clock) and let it fully process everything, in case there is a backlog of invoices to generate.

**Comment by karser** [comment]
I restarted the KB instance (down && up), but the issue still persists.

>Are you sure you don't have old data laying around?

I do have old data under other tenants and apparently, that's why it is failing. 

Sorry for being annoying, but if the same happens on production it's not that easy to drop the database and fix this issue. This might be a KB bug.

**Comment by pierre** [comment]
> I do have old data under other tenants and apparently, that's why it is failing.

Did you check the Kill Bill logs to understand why the invoice isn't generated?

> Sorry for being annoying, but if the same happens on production it's not that easy to drop the database and fix this issue. This might be a KB bug.

Happy to investigate further if you think there is a bug, but I'm unable to reproduce the issue.

**Comment by karser** [comment]
Here is how to reproduce this issue:

1. Download and untar the mysql data directory:
```bash
curl https://s3.eu-west-3.amazonaws.com/killbill-test/mysql.tgz | tar xvz
```

2. bind-mount this directory to `/var/lib/mysql`
```yaml
    db:
        image: killbill/mariadb:0.20
        volumes:
            - "./mysql:/var/lib/mysql"
```
3. `docker compose down && docker compose up -d`
4. vendor/bin/phpunit -c phpunit.xml test/ServerInvoiceTest.php

**Comment by karser** [comment]
And here is the log: https://s3.eu-west-3.amazonaws.com/killbill-test/killbill.out

I noticed an error in the log: `Unknown column 'is_active' in 'field list'`.

I tried running `$MIGRATIONS_CMD` but it didn't help. And there shouldn't be changes in DB structure because it has always been v0.20.10.

**Comment by pierre** [comment]
It looks like you have an old version of `killbill/mariadb:0.20`. Could you delete your local image and volume, and try again?

**Comment by karser** [comment]
Thanks, already did that.
My concern if this happens on production instance it wouldn't be so easy to fix it by dropping the db. Looks like the database went to the inconsistent state, although I've been staying all the time on v0.20.10.
Anyways, I hope this will never happen again, I'll be looking in the KB logs more closely.

**Comment by pierre** [comment]
> My concern if this happens on production instance it wouldn't be so easy to fix it by dropping the db. Looks like the database went to the inconsistent state, although I've been staying all the time on v0.20.10.

My guess is that your setup was incorrect from the beginning but it somehow didn't impact other tests.

Kill Bill is almost 10 years old now, and it has been many years since a critical bug was reported. Not saying there aren't any bug anymore, but we are highly confident the software is stable and production ready. It should not suddenly _go to an inconsistent state_. 😺 

> I'll be looking in the KB logs more closely.

This is good practice regardless, to make sure the system is behaving correctly.

**Pull request merged**